sdk/core/Azure.Core/src/RequestOptions.cs

```diff
@@ -0,0 +1,146 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using Azure.Core;
+using Azure.Core.Pipeline;
+
+namespace Azure
+{
+    /// <summary>
+    /// Options that can be used to control the behavior of a request sent by a client.
+    /// </summary>
+    public class RequestOptions
+    {
+        private bool _frozen;
```

Comment:
Should we consider making this `volatile` to ensure that all threads see a consistent value?

Reply:
Considered! :)  We won't make this `volatile` because we don't support mutating `RequestOptions` on multiple threads.  `_frozen` guards against the situation where someone would try to mutate the RO after using it in a method call.